fedimint-client/src/lib.rs

```diff
@@ -289,13 +288,14 @@ pub trait IGlobalClientContext: Debug + MaybeSend + MaybeSync + 'static {
     fn decoders(&self) -> &ModuleDecoderRegistry;
 
     /// This function is mostly meant for internal use, you are probably looking
-    /// for [`DynGlobalClientContext::claim_input`].
+    /// for [`DynGlobalClientContext::claim_inputs`].
     /// Returns transaction id of the funding transaction and an optional
     /// `OutPoint` that represents change if change was added.
-    async fn claim_input_dyn(
+    async fn claim_inputs_dyn(
         &self,
         dbtx: &mut ClientSMDatabaseTransaction<'_, '_>,
-        input: InstancelessDynClientInput,
+        inputs: Vec<InstancelessDynClientInput>,
+        input_sms: Vec<InstancelessDynClientInputSM>,
```

Comment:
Having the SMs be part of the input type was quite intentional: I don't want developers to be able to claim an input without spawning the required SMs. Maybe we could just ceate an `InstancelessDynClientInputs` type that contains a list of inputs and a list of SMs? That way users can't accidentally mess up. 

Just imagine someone getting the happy path to work by just handing in an empty vec and then forgetting to pass through the SMs.

Reply:
Done in #6207